--- ./docker/mongodb/docker-entrypoint.sh	2022-11-23 11:21:07.329753300 +0700
+++ ./docker/mongodb/docker-entrypoint-patched.sh	2022-11-23 12:24:15.039206200 +0700
@@ -299,6 +299,10 @@
 	fi
 
 	if [ -n "$shouldPerformInitdb" ]; then
+	
+        openssl rand -base64 756 > /etc/mongo-key/mongodb.key
+        chmod 400 /etc/mongo-key/mongodb.key
+
 		mongodHackedArgs=( "$@" )
 		if _parse_config "$@"; then
 			_mongod_hack_ensure_arg_val --config "$tempConfigFile" "${mongodHackedArgs[@]}"
@@ -419,7 +423,14 @@
 		set -- "$@" --bind_ip_all
 	fi
 
-	unset "${!MONGO_INITDB_@}"
+    echo 'Initiating replica set'
+    "$@" --logpath "/proc/$$/fd/1" --fork
+    echo 'rs.initiate({"_id":"rs","members":[{"_id":0,"host":"m0:27017"},{"_id":1,"host":"m1:27017"},{"_id":2,"host":"m2:27017"}]});' | mongosh -u "$MONGO_INITDB_ROOT_USERNAME" -p "$MONGO_INITDB_ROOT_PASSWORD"
+    "$@" --logpath "/proc/$$/fd/1" --shutdown
+    echo 'Done initiating replica set'
+
+    unset "${!MONGO_INITDB_@}"
+
 fi
 
 rm -f "$jsonConfigFile" "$tempConfigFile"
